src/components/keymap/keymap.component.ts

```diff
@@ -0,0 +1,138 @@
+import { Component, ViewChildren, QueryList, ElementRef, OnInit, AfterViewInit, Renderer } from '@angular/core';
+import { ActivatedRoute } from '@angular/router';
+
+import { SvgKeyboardPopoverComponent } from '../svg-keyboard-popover.component';
+import { Layers } from '../../../config-serializer/config-items/Layers';
+import { UhkConfigurationService } from '../../services/uhk-configuration.service';
+
+@Component({
+    selector: 'keymap',
+    template: require('./keymap.component.html'),
+    styles: [require('./keymap.component.scss')],
+    directives: [SvgKeyboardPopoverComponent],
+    providers: [UhkConfigurationService]
+})
+export class KeymapComponent implements OnInit, AfterViewInit {
+    @ViewChildren('baseButton,modButton,fnButton,mouseButton')
+    buttonsQueryList: QueryList<ElementRef>;
+
+    @ViewChildren(SvgKeyboardPopoverComponent, { read: ElementRef })
+    keyboardsQueryList: QueryList<ElementRef>;
+
+    private buttons: ElementRef[];
+    private keyboards: ElementRef[];
+    private selectedLayerIndex: number;
+    private keymapId: number = 0;
+
+    private layers: Layers;
+    private keymap: any;
```

Comment:
Change `any` to `Keymap` imported from `config-serializer/config-items/Keymap.ts`.